### PR TITLE
Add metrics instrumentation outline to feature serve

### PR DESCRIPTION
**Pull request by Sami1309**

### CODE REVIEW

File: ff/serving/csv.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ func (provider *LocalCSVProvider) GetDatasetReader(key map[string]string) (Datas
 
 func (provider *LocalCSVProvider) ToKey(path string, schema CSVSchema) map[string]string {
 	key := make(map[string]string)
+	logger := provider.Logger.With("Key", key)
```
Comment (simba-git) [inline ff/serving/csv.go:48]: This doesn't do anything, it logs an empty map

File: ff/serving/serving.go [hunk 2]
```diff
@@ -54,40 +57,50 @@ func NewTrainingDataServer(logger *zap.SugaredLogger) (*TrainingDataServer, erro
 func (serv *TrainingDataServer) TrainingData(req *pb.TrainingDataRequest, stream pb.OfflineServing_TrainingDataServer) error {
 	id := req.GetId()
 	name, version := id.GetName(), id.GetVersion()
+	featureObserver := prom_metrics.BeginObservingFeatureServe(name+" "+version, "training data")
```
Comment (simba-git) [inline ff/serving/serving.go:60]: run go fmt

Comment (simba-git) [reply ff/serving/serving.go:60]: what happens if name has a space in it or if version has a space in it. Also why make up a string called "training data", not good for to force users to memorize a fixed set of strings that metrics actually uses on the frontend, Try using a set of constants

File: ff/serving/serving.go [hunk 3]
```diff
@@ -54,40 +57,50 @@ func NewTrainingDataServer(logger *zap.SugaredLogger) (*TrainingDataServer, erro
...
 		serv.Logger.Errorw("Dataset error", "Error", err)
+		featureObserver.SetError()
 		return err
 	}
 	return nil
 }
 
 func main() {
 	logger := zap.NewExample().Sugar()
+	prom_metrics = metrics.NewMetrics("test")
```
Comment (simba-git) [inline ff/serving/serving.go:101]: What's name used for?

File: metrics/metrics_service_example.go [hunk 4]
```diff
@@ -10,7 +10,10 @@ var metrics MetricsHandler
 func testServing() {
 	go func() {
 		for {
+			serveFeature("fixed_acidity", "feature_key")
 			serveFeature("Non-free Sulfur Dioxide", "feature_key")
+			serveFeature("Wine quality set", "Feature Set")
+
```
Comment (simba-git) [inline metrics/metrics_service_example.go:16]: go fmt

File: metrics/interface.go [hunk 5]
```diff
@@ -1,6 +1,8 @@
 package main
 
 import (
+	//remote packages
```
Comment (simba-git) [inline metrics/interface.go:4]: ?

**Review by simba-git — CHANGES_REQUESTED** [review]

File: ff/serving/serving.go (hunk 3 above)
Comment (Sami1309) [reply ff/serving/serving.go:101]: A label specifying the instance of metrics observation. In practice should be an environment variable tied to an instance so metrics can be queries based on belonging to an instance or super-set type structure.

**Review by shabbyjoon — APPROVED** [review]

**Pull request merged**